cirq/google/optimizers/convert_to_sqrt_iswap.py

```diff
@@ -0,0 +1,290 @@
+        turns: Exponent specifying the evolution time in number of rotations.
+    """
+    theta = (turns % 2) * np.pi
+    if 0 <= theta <= np.pi:
+        sign = 1.
+        theta_prime = theta
+    elif np.pi < theta < 2 * np.pi:
+        sign = -1.
+        theta_prime = 2 * np.pi - theta
+
+    if np.isclose(theta, np.pi):
+        # If we are close to pi, just set values manually to avoid possible
+        # numerical errors with arcsin of greater than 1.0 (Ahem, Windows).
+        phi = np.pi / 2
+        xi = np.pi / 2
+    else:
+        phi = np.arcsin(np.sqrt(2) * np.sin(theta_prime / 4))
+        xi = np.arctan(np.tan(phi) / np.sqrt(2))
+
+    yield cirq.Rz(sign * 0.5 * theta_prime).on(a)
+    yield cirq.Rz(sign * 0.5 * theta_prime).on(b)
+    yield cirq.Rx(xi).on(a)
+    yield cirq.X(b)**(-sign * 0.5)
+    yield SQRT_ISWAP(a, b)
+    yield cirq.Rx(-2 * phi).on(a)
+    yield cirq.Z(a)
+    yield SQRT_ISWAP(a, b)
+    yield cirq.Z(a)
+
+    yield cirq.Rx(xi).on(a)
+    yield cirq.X(b)**(sign * 0.5)
+    # Corrects global phase
+    yield cirq.GlobalPhaseOperation(np.exp(sign * theta_prime * 0.25j))
+
+
+def cphase_symbols_to_sqrt_iswap(a, b, turns):
+    """Version of cphase_to_sqrt_iswap that works with symbols.
+
+    Note that the formulae contained below will need to be flattened
+    into a sweep before serializing.
+    """
+    theta = sympy.Mod(turns, 2.0) * sympy.pi
+
+    # -1 if theta > pi.  Adds a hacky fudge factor so theta=pi is not 0
+    sign = sympy.sign(sympy.pi - theta + 1e-9)
+
+    # For sign = 1: theta. For sign = -1, 2pi-theta
+    theta_prime = (sympy.pi - sign * sympy.pi) + sign * theta
+
+    phi = sympy.asin(np.sqrt(2) * sympy.sin(theta_prime / 4))
+    xi = sympy.atan(sympy.tan(phi) / np.sqrt(2))
+
+    yield cirq.Rz(sign * 0.5 * theta_prime).on(a)
+    yield cirq.Rz(sign * 0.5 * theta_prime).on(b)
+    yield cirq.Rx(xi).on(a)
+    yield cirq.X(b)**(-sign * 0.5)
+    yield SQRT_ISWAP(a, b)
+    yield cirq.Rx(-2 * phi).on(a)
+    yield cirq.Z(a)
+    yield SQRT_ISWAP(a, b)
+    yield cirq.Z(a)
+    yield cirq.Rx(xi).on(a)
+    yield cirq.X(b)**(sign * 0.5)
+
+
+def iswap_to_sqrt_iswap(a, b, turns):
+    """Implement the evolution of the hopping term using two sqrt_iswap gates
+     and single-qubit operations. Output unitary:
+    [1   0   0   0],
+    [0   c  is   0],
+    [0  is   c   0],
+    [0   0   0   1],
+    where c = cos(t * np.pi / 2) and s = sin(t * np.pi / 2).
+
+    Args:
+        a: the first qubit
+        b: the second qubit
+        t: Exponent that specifies the evolution time in number of rotations.
+    """
+    yield cirq.Z(a)**0.75
+    yield cirq.Z(b)**0.25
+    yield SQRT_ISWAP(a, b)
+    yield cirq.Z(a)**(-turns / 2 + 1)
+    yield cirq.Z(b)**(turns / 2)
+    yield SQRT_ISWAP(a, b)
+    yield cirq.Z(a)**0.25
+    yield cirq.Z(b)**-0.25
+
+
+def swap_to_sqrt_iswap(a, b, turns):
+    r"""Implement the evolution of the hopping term using two sqrt_iswap gates
```

Comment:
nit: does this need to be a raw string? I would think that's only necessary if there are literal backslashes, which isn't the case here.

Reply:
Done.